servicetalk-opentelemetry-asynccontext/src/main/java/io/servicetalk/opentelemetry/asynccontext/ServiceTalkContextStoreProvider.java

```diff
@@ -1,95 +0,0 @@
-/*
- * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package io.servicetalk.opentelemetry.asynccontext;
-
-import io.servicetalk.concurrent.api.AsyncContext;
-import io.servicetalk.context.api.ContextMap;
-
-import io.opentelemetry.context.Context;
-import io.opentelemetry.context.ContextStorage;
-import io.opentelemetry.context.ContextStorageProvider;
-import io.opentelemetry.context.Scope;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.lang.invoke.MethodHandles;
-import javax.annotation.Nullable;
-
-import static io.servicetalk.context.api.ContextMap.Key.newKey;
-
-/**
- * Implementation of {@link ContextStorageProvider} that stores the Tracing Context
- * making it available within {@link AsyncContext}.
- */
-public final class ServiceTalkContextStoreProvider implements ContextStorageProvider {
```

Comment:
Do we need to keep this around? I think it's technically public API, but it's not intended to be used other than via service loading. We could simply keep the class and delete the `services` file, or we could keep the class and make its constructor throw an exception saying "don't use this anymore".

Reply:
I couldn't find anyone using it directly, which matches your expectations. I'm fine with removing it, we can add it back in a follow-up release if users complain.